### ISSUE TITLE
shinyapp deployment error | themes

### ISSUE THREAD

**Issue by antoine4ucsd** [JohnCoene/echarts4r#587]
Hello
Not sure if there is an easy fix but I created an app that works just fine locally with several nice echarts plot
the deployment went welll but the app is not running online (error)
I checked the log:

```
Error in value[[3L]]: Couldn't normalize path in `addResourcePath`, with arguments: `prefix` = 'shine-1.0.0'; `directoryPath` = '/Users/me/Library/R/x86_64/4.2/library/echarts4r/htmlwidgets/lib/echarts-4.8.0/themes'
```

is there a fix? it may have been addressed elsewhere but cannot find it.
I used shine theme for most plots

thank you!

**Comment by JohnCoene** [comment]
I think that was a bug in {htmlwidgets}, can you reinstall it and try again? Is it via shinyapps.io?

```r
install.packages("htmlwidgets")
```

**Comment by antoine4ucsd** [comment]
trying now. thank you!

**Comment by antoine4ucsd** [comment]
I added  the above line in my code and specifically call the htmlwidgets library but this is still looking for a local path 
not sure how to handle this. really sorry
```
2023-11-08T08:14:46.657309+00:00 shinyapps[10454283]: Warning: Error in value[[3L]]: Couldn't normalize path in `addResourcePath`, with arguments: `prefix` = 'shine-1.0.0'; `directoryPath` = '/Users/me/Library/R/x86_64/4.2/library/echarts4r/htmlwidgets/lib/echarts-4.8.0/themes'

```

**Comment by antoine4ucsd** [comment]
attaching the 2 logs with and without the install.packages within the code.
[miami_dashboard-logs.txt](https://github.com/JohnCoene/echarts4r/files/13294288/miami_dashboard-logs.txt)
[miami2-logs.txt](https://github.com/JohnCoene/echarts4r/files/13294291/miami2-logs.txt)

**Comment by JohnCoene** [comment]
Where do you deploy this app?

The paths in there seems to be local paths on Windows machines...

**Comment by antoine4ucsd** [comment]
I agree . I am deploying the app from my laptop but it is submitted to shinyapps.io. I think I understand why now... I saved outputs that are called directly within the apps. bummer. I will have to go with a different strategy I guess. I will start from scratch and keep you posted. sorry for wasting your time. thank you!

**Comment by JohnCoene** [comment]
No worries but why are you storing plots locally? You should not need to do that, and even if you do you can make it work.

Use relative paths instead of normalised hardcoded path.

```
# good
./img.png

# bad
/User/me/img.png
```

**Comment by antoine4ucsd** [comment]
I know. I am not storing the plot itself but the RDS of the echart output. I think this is a stupid habit...

**Comment by antoine4ucsd** [comment]
I confirm that was it! may I ask one more thing...
is there a way to increase the height of the echarts4rOutput generated with render renderEcharts4r?
see for example:

![image](https://github.com/JohnCoene/echarts4r/assets/18502631/fc232803-2aa8-4c79-ae51-b1f3a46192ec)

I can increase the size of the box but not the plot itself.
thank you for your patience.

**Comment by JohnCoene** [comment]
Yes here: https://echarts4r.john-coene.com/reference/echarts4r-shiny

**Comment by antoine4ucsd** [comment]
wahoo. I just tried: I confirm it works. sorry for keeping you busy but it looks like nothing is impossible so far.
thank you!

**Comment by aarong1** [comment]
Hi - I understand the use case. the ticket referred to building the plot off line and shipping the echarts plot object with the app and just reading it in the app. this causes problems with the paths to where the themes are stored.  Is there a way to redefine the path the themes are stored at, or make it use ```system.file(package = "echarts4r")``` to find the path. an excerpt of my - very similar error is below, deploying from a mac to linux (posit connect)

`2026-07-21T20:43:50+01:00 Warning: Error in value[[3L]]: Couldn't normalize path in 'addResourcePath', with arguments: `prefix` = 'auritus-1.0.0'; `directoryPath` = '/Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/library/echarts4r/htmlwidgets/lib/echarts-6.0.0/themes'
`
`print(system.file(package = "echarts4r"))` in the deployed app environment is

2026-07-21T20:43:41+01:00 [1] "/cloud/lib/x86_64-pc-linux-gnu-library/4.5/echarts4r"

**Comment by JohnCoene** [comment]
can you try to completely uninstall echarts4r and do a clean install. The path reported seems correct to me so it should work
